clients/javascript/model/game.js

```diff
@@ -0,0 +1,105 @@
+const Level = require('./level').Level;
+const Properties = require('./properties').Properties;
+const Player = require('./player').Player;
+const Unit = require('./unit').Unit;
+const Bullet = require('./bullet').Bullet;
+const Mine = require('./mine').Mine;
+const LootBox = require('./loot-box').LootBox;
+
+class Game {
+    constructor (currentTick, properties, level, players, units, bullets, mines, lootBoxes) {
+        this.currentTick = currentTick;
+        this.properties = properties;
+        this.level = level;
+        this.players = players;
+        this.units = units;
+        this.bullets = bullets;
+        this.mines = mines;
+        this.lootBoxes = lootBoxes;
+    }
+
+    static async readFrom (stream) {
+        const currentTick = await stream.readInt();
+        const properties = await Properties.readFrom(stream);
+        const level = await Level.readFrom(stream);
+
+        let i, arraySize;
```

Comment:
Probably here i should duplicate **i** and **arraySize** to make it code generator friendly